packages/api/src/opamp/controllers/opampController.ts

```diff
@@ -0,0 +1,203 @@
+
+import { getTeam } from '@/controllers/team';
+import logger from '@/utils/logger';
+
+import { agentService } from '../services/agentService';
+import {
+  createRemoteConfig,
+  decodeAgentToServer,
+  encodeServerToAgent,
+  serverCapabilities,
+} from '../utils/protobuf';
+
+type CollectorConfig = {
+  extensions: Record<string, any>;
+  receivers: {
+    'otlp/hyperdx'?: {
+      protocols: {
+        grpc: {
+          endpoint: string;
+          include_metadata: boolean;
+          auth?: {
+            authenticator: string;
+          };
+        };
+        http: {
+          endpoint: string;
+          cors: {
+            allowed_origins: string[];
+            allowed_headers: string[];
+          };
+          include_metadata: boolean;
+          auth?: {
+            authenticator: string;
+          };
+        };
+      };
+    };
+    nop?: null;
+  };
+  service: {
+    extensions: string[];
+    pipelines: {
+      [key: string]: {
+        receivers: string[];
+      };
+    };
+  };
+};
+export class OpampController {
+  /**
+   * Handle an OpAMP message from an agent
+   */
+  public async handleOpampMessage(req: Request, res: Response): Promise<void> {
+    try {
+      // Check content type
+      const contentType = req.get('Content-Type');
+      if (contentType !== 'application/x-protobuf') {
+        res
+          .status(415)
+          .send(
+            'Unsupported Media Type: Content-Type must be application/x-protobuf',
+          );
+        return;
+      }
+
+      // Decode the AgentToServer message
+      const agentToServer = decodeAgentToServer(req.body);
+      logger.debug('agentToServer', agentToServer);
+      logger.debug(
+        // @ts-ignore
+        `Received message from agent: ${agentToServer.instanceUid?.toString(
+          'hex',
+        )}`,
+      );
+
+      // Process the agent status
+      const agent = agentService.processAgentStatus(agentToServer);
+
+      // Prepare the response
+      const serverToAgent: any = {
+        instanceUid: agent.instanceUid,
+        capabilities: serverCapabilities,
+      };
+
+      // Check if we should send a remote configuration
+      if (agentService.agentAcceptsRemoteConfig(agent)) {
+        const team = await getTeam();
+        // This is later merged with otel-collector/config.yaml
+        // we need to instantiate a valid config so the collector
+        // can at least start up
+        const NOP_CONFIG: CollectorConfig = {
+          extensions: {},
+          receivers: {
+            nop: null,
+          },
+          service: {
+            extensions: [],
+            pipelines: {
+              traces: {
+                receivers: ['nop'],
+              },
+              metrics: {
+                receivers: ['nop'],
+              },
+              'logs/in': {
+                receivers: ['nop'],
+              },
+            },
+          },
+        };
+        let config = NOP_CONFIG;
```

Comment:
style: we can just init `config` without assigning another var

Reply:
this is a bit more explicit imo